### PR TITLE
Bugfix- LOA3 multifactor users were failing validation

### PR DESCRIPTION
**Pull request by annaswims**
## Description of change
We were improperly mapping `myhealthevet_loa3`  and `dslogon_multifactor` attributes to id.me, leading to nil attributes and therefore an invalid LOA3 user.

Fixes [this sentry error](http://sentry.vetsgov-internal/vets-gov/platform-api-production/issues/29156/events/2122251/) (consider data after the 2/8 relase valid for the sake of this fix.)

## Acceptance Criteria (Definition of Done)
we should no longer see any "Login Fail! on User/Session Validation" errors in sentry. 
we should see  `myhealthevet_loa3`  and `dslogon_multifactor` on the graph for [Success rate for saml_callback by context](http://grafana.vetsgov-internal/dashboard/db/site-authentication?panelId=7&fullscreen&orgId=1&from=now-7d&to=now)

### CODE REVIEW

File: lib/saml/user.rb [hunk 1]
```diff
@@ -79,13 +79,11 @@ def authn_context
...
     def user_attributes_class
       case authn_context
-      when 'myhealthevet'; then SAML::UserAttributes::MHV
-      when 'dslogon'; then SAML::UserAttributes::DSLogon
-      when 'myhealthevet_multifactor', 'dslogon_multifactor', 'multifactor'
-        SAML::UserAttributes::IdMe
-      when 'dslogon_loa3', 'myhealthevet_loa3', LOA::IDME_LOA3
-        SAML::UserAttributes::IdMe
-      when LOA::IDME_LOA1
+      when 'myhealthevet', 'myhealthevet_multifactor'
```
Comment (saneshark) [inline lib/saml/user.rb]: I don't think this is accurate. The payload returned for myhealthevet_multifactor is entirely made up of ID.me attributes.

File: lib/saml/user.rb [hunk 2]
```diff
@@ -79,13 +79,11 @@ def authn_context
...
+        SAML::UserAttributes::MHV
+      when 'dslogon', 'dslogon_multifactor'
```
Comment (saneshark) [inline lib/saml/user.rb]: same comment here... 'dslogon' includes attributes from both ID.me and DS Logon, but `dslogon_multifactor` is identical to `multifactor`, it's only ID.me attributes and should not be treated as a DS Logon type of user attributes class.

**Comment by saneshark** [comment]
What we need to do instead, is to make sure that the new_user that is persisted with the existing user's attributes and that we are only changing the multifactor attribute to true, based on the new saml response.

It's possible that we're attempting to initialize a user differently than that, and that is why it is failing validation, but the saml responses mapping to attribute classes are not the problem.

**Comment by annaswims** [comment]
I'm relatively confident that this will help.  Take a look at the saml_attributes for  http://sentry.vetsgov-internal/vets-gov/platform-api-production/issues/29156/.  The validation errors are mostly for dslogon_multifactor and mhv_multifactor LOA3 users. 

Combined with the near 0% success rate here:
http://grafana.vetsgov-internal/dashboard/db/site-authentication?panelId=16&fullscreen&orgId=1&from=now-7d&to=now

**Comment by saneshark** [comment]
It just doesn't make any sense for this to make any affect. Can we hop on a call? 

The issues with user validation is in SSOService, here we are only concerned with mapping the attributes we receive to an appropriate attribute parsing class. The attributes that we receive for multifactor are identical to those received for

**Pull request merged**